<div style="text-align: center;">
<div style="font-size: 18px;display:inline-block; padding:8px 16px; border:2px solid #7e57c2; border-radius:8px; background-color:#f5f0ff; font-family:'Segoe UI', sans-serif;">
  <h1 style="margin:0; color:#6a1b9a;">🌍 Impact of AI on Digital Media (2020-2025)</h1>
</div>
</div>

### <B><I> "If AI is the electricity of the 21st century — which countries are lighting  up the world, and 
### <B><I>  which are still in the blackout zone?"

<img src="ai-tools.webp" alt="Image" style="width:1200px; height:500px;"/>

📦 **Source:** Kaggle – [🌍 Impact of AI on Digital Media (2020-2025)](https://www.kaggle.com/datasets/atharvasoundankar/impact-of-ai-on-digital-media-2020-2025)


<h2 style="color:#0D47A1;">🧭 Table of Contents</h2>

<ul style="font-size:16px; line-height:1.7;">
  <li><strong>1. 📘 Project Overview & Dataset Introduction</strong></li>
  <li><strong>2. 🧼 Data Exploring & Cleaning </strong></li>
  <li><strong>3. 📊 Advanced SQL Insights</strong>
    <ul>
      <li>💥 AI Impact Power Index by Country</li>
      <li>⚠️ Identify Industries with High AI Adoption but Low Economic Return</li>
      <li>🔁 Rank the Most Volatile Industry-Country Pairs in AI Adoption</li>
      <li>📈 Identify Early Adopters (Industries Above Global Average AI Adoption in 2020)</li>
      <li>🏆 Year-over-Year Top Performing Country by Economic Gain per AI Unit</li>
      <li>🧠 AI Maturity vs Market Presence</li>
      <li>🔐 AI Content Creation vs Economic and Workforce Impact</li>
      <li>🧮 Identifying Significant Year-over-Year (YoY) Surges in AI Adoption</li>
      <li>📉 AI Adoption and Job Loss Paradox: Industries with High Adoption but Decreasing Job Loss</li>
      <li>📊 AI Tool Transitions Driving Business Impact Across Industries</li>
      <li>⚖️ AI Content Volume vs Regulatory Readiness</li>
      <li>🎯 Trust & Market Share Growth Buckets</li>
      <li>🎯 AI Consistency Champions: Country–Industry Pairs with Multi-Year Performance Above Global Standards</li>
    </ul>
  </li>
  <li><strong>4. ✅ Final Summary & Creative Conclusion</strong></li>
</ul>


<div style="background-color:#f5f0ff; padding:14px; border-left:6px solid #7e57c2; font-family:'Segoe UI', sans-serif; font-size:15px; line-height:1.6;">
  <b style="font-size:18px; color:#6a1b9a;">🎯 Project Objective</b><br><br>
  This project explores the global AI evolution between <b>2020 and 2025</b>, decoding which countries and industries are truly leveraging AI to lead the digital economy.<br><br>
  🔍 <b style="color:#7e57c2;">Key Goals:</b>
  <ul style="margin-left:-20px;">
    <li>🌍 Analyze how AI adoption varies across industries and nations</li>
    <li>📈 Discover the link between AI and measurable business outcomes like revenue and collaboration</li>
    <li>🤖 Identify the most trusted AI tools and understand their impact on consumer perception</li>
    <li>⚖️ Explore the role of regulation, trust, and content creation in shaping AI maturity</li>
    <li>🏆 Highlight consistent leaders and uncover hidden patterns using advanced SQL insights</li>
  </ul>
</div>

<div style="background-color:#e8f4fd; padding:12px; border-left:6px solid #1f77b4;">
  <b style="font-size:18px; color:#1f77b4;">📘 Why this project?</b><br>
  In a world where AI is reshaping industries, this project dives into the digital heartbeat of 2020–2025. It explores which countries are leading the AI revolution, which tools inspire the most trust, and how industries are navigating the balance between automation and human collaboration.
</div>


<div style="background-color:#e9fbe7; padding:14px; border-left:6px solid #4caf50; font-family:'Segoe UI', sans-serif; font-size:15px; line-height:1.6;">
  <b style="font-size:18px; color:#2e7d32;">📊 About the Dataset</b><br><br>
  This dataset explores the influence of <b>AI-generated content</b> across various industries, including <i>Gaming, media, education, and Healthcare etc.</i>. It provides a multi-faceted view into public sentiment, engagement trends, economic impact, and global regulatory responses from <b>2020 to 2025</b>.<br><br>

  As AI becomes more embedded in everyday digital experiences, this dataset serves as a <b>valuable resource</b> for data analysts, business strategists, and machine learning researchers to:
  <ul style="margin-left:-20px;">
    <li>🔍 Study adoption trends and behavioral shifts</li>
    <li>📉 Detect risks like bias or job loss signals</li>
    <li>📈 Predict future AI usage and market readiness</li>
  </ul>
</div>


<!-- End of Previous Section -->
<hr style="border: none; border-top: 2px dashed #ccc; margin: 30px 0;" />

<div style="background-color:#e0f7f4; padding:10px; border-left:6px solid #26a69a; font-family:'Segoe UI', sans-serif;">
  <h2 style="margin:0; color:#00695c;">📦 Importing Libraries and Setup</h2>
</div>

<p style="font-family:'Segoe UI', sans-serif; font-size:15px; line-height:1.6;">
  Let's start by importing essential Python libraries and configuring the environment for data analysis and SQL integration.
</p>

In [1]:
import pandas as pd
from sqlalchemy import create_engine

# 🔑 MySQL connection parameters
username = 'root'          
password = 'password' 
host = 'localhost'         
port = 3306               
database = 'global_ai'   

# 🔌 Create SQLAlchemy engine to connect MySQL to Python
engine = create_engine(f'mysql+pymysql://{username}:{password}@{host}:{port}/{database}')

<!-- End of Previous Section -->
<hr style="border: none; border-top: 2px dashed #ccc; margin: 30px 0;" />

<div style="background-color:#fce4ec; padding:10px; border-left:6px solid #ec407a; font-family:'Segoe UI', sans-serif;">
  <h2 style="margin:0; color:#ad1457;">🔍 Data Exploration</h2>
</div>

<p style="font-family:'Segoe UI', sans-serif; font-size:15px; line-height:1.6;">
  Exploring key variables, data distributions, and initial patterns to understand the structure and potential of the dataset before deep analysis.
</p>

In [2]:
# Displaying the first 5 rows of the dataset to get an initial look

query = "SELECT * FROM ai LIMIT 5"
pd.read_sql(query, engine)

Country  Year    Industry  AI Adoption Rate (%)  \
0  South Korea  2022       Media                 44.29   
1        China  2025       Legal                 34.75   
2          USA  2022  Automotive                 81.06   
3       France  2021       Legal                 85.24   
4       France  2021      Gaming                 78.95   

   AI-Generated Content Volume (TBs per year)  Job Loss Due to AI (%)  \
0                                       33.09                   16.77   
1                                       66.74                   46.89   
2                                       96.13                   10.66   
3                                       93.76                   27.70   
4                                       45.62                   17.45   

   Revenue Increase Due to AI (%)  Human-AI Collaboration Rate (%)  \
0                           46.12                            74.79   
1                           52.46                            26.17   
2                           45.60                            39.66   
3                           78.24                            29.45   
4                            1.05                            21.70   

  Top AI Tools Used Regulation Status  Consumer Trust in AI (%)  \
0              Bard            Strict                     40.77   
1            DALL-E            Strict                     35.67   
2  Stable Diffusion          Moderate                     54.47   
3            Claude          Moderate                     51.84   
4        Midjourney            Strict                     41.77   

   Market Share of AI Companies (%)  
0                             18.73  
1                             35.02  
2                             22.76  
3                              1.93  
4                             21.41

In [3]:
# Displaying the list of unique countries present in the dataset

query = "SELECT DISTINCT(Country) FROM ai"
pd.read_sql(query, engine)

Country
0  South Korea
1        China
2          USA
3       France
4    Australia
5           UK
6       Canada
7        India
8        Japan
9      Germany

In [4]:
# Displaying the list of unique Industries present in the dataset

query = "SELECT DISTINCT(Industry) FROM ai"
pd.read_sql(query, engine)

Industry
0          Media
1          Legal
2     Automotive
3         Gaming
4         Retail
5      Education
6     Healthcare
7      Marketing
8  Manufacturing
9        Finance

In [5]:
# Displaying the list of unique Top AI Tools Used present in the dataset

query = "SELECT DISTINCT `Top AI Tools Used` FROM ai"
pd.read_sql(query, engine)

Top AI Tools Used
0              Bard
1            DALL-E
2  Stable Diffusion
3            Claude
4        Midjourney
5           ChatGPT
6         Synthesia

In [6]:
# Displaying the list of unique year present in the dataset

query = "SELECT DISTINCT(year) FROM ai"
pd.read_sql(query, engine)

year
0  2022
1  2025
2  2021
3  2023
4  2020
5  2024

In [7]:
query = "SELECT * FROM ai WHERE country = 'France' AND Industry = 'Media' ORDER BY Year "
pd.read_sql(query, engine)

Country  Year Industry  AI Adoption Rate (%)  \
0  France  2020    Media                 20.47   
1  France  2020    Media                 55.19   
2  France  2021    Media                 42.53   
3  France  2022    Media                 49.97   
4  France  2023    Media                 88.41   
5  France  2024    Media                 23.27   
6  France  2025    Media                 79.64   

   AI-Generated Content Volume (TBs per year)  Job Loss Due to AI (%)  \
0                                        1.61                    2.43   
1                                       41.06                   29.26   
2                                       44.38                    8.99   
3                                        5.59                   22.61   
4                                        5.34                   36.07   
5                                       38.75                   44.22   
6                                       87.25                   12.32   

   Revenue Increase Due to AI (%)  Human-AI Collaboration Rate (%)  \
0                           31.04                            52.49   
1                           51.90                            49.04   
2                           49.69                            47.11   
3                           17.55                            34.97   
4                           13.78                            36.24   
5                           67.88                            78.13   
6                           54.75                            82.48   

  Top AI Tools Used Regulation Status  Consumer Trust in AI (%)  \
0  Stable Diffusion            Strict                     35.80   
1           ChatGPT          Moderate                     50.12   
2         Synthesia          Moderate                     55.15   
3         Synthesia           Lenient                     78.71   
4         Synthesia          Moderate                     58.85   
5           ChatGPT          Moderate                     72.79   
6            DALL-E          Moderate                     49.54   

   Market Share of AI Companies (%)  
0                             24.29  
1                             46.08  
2                             46.58  
3                             27.60  
4                             35.86  
5                             13.50  
6                             20.17

<div style="background-color:#f5f0ff; padding:14px; border-left:6px solid #7e57c2; font-family:'Segoe UI', sans-serif; font-size:15px; line-height:1.6;">
  <i>Here for country = "France" and Industry = "Media" for 6 years [2020-2025], we have 7 entries as for year 2020 we have entries for 2 different AI's.</i>
</div>

<!-- End of Previous Section -->
<hr style="border: none; border-top: 2px dashed #ccc; margin: 30px 0;" />

<div style="background-color:#fff3e0; padding:10px; border-left:6px solid #fb8c00; font-family:'Segoe UI', sans-serif;">
  <h2 style="margin:0; color:#e65100;">🧪 Data Cleaning</h2>
</div>

<p style="font-family:'Segoe UI', sans-serif; font-size:15px; line-height:1.6;">
  Removing duplicates, handling missing values to prepare for meaningful analysis.
</p>

<div style="background-color:#e6f4ea; padding:12px; border-left:5px solid #2e7d32; font-family:'Segoe UI', sans-serif;">
  <h4>🔹Identify NULLs in key columns:
</div>


In [8]:
# Count NULLs or empty strings in each column
query = """
SELECT
    SUM(CASE WHEN Country IS NULL OR Country = '' THEN 1 ELSE 0 END) AS missing_country,
    SUM(CASE WHEN Year IS NULL OR Year = '' THEN 1 ELSE 0 END) AS missing_year,
    SUM(CASE WHEN Industry IS NULL OR Industry = '' THEN 1 ELSE 0 END) AS missing_industry,
    SUM(CASE WHEN `Top AI Tools Used` IS NULL OR `Top AI Tools Used` = '' THEN 1 ELSE 0 END) AS missing_top_ai_tool_used,
    SUM(CASE WHEN `AI Adoption Rate (%%)` IS NULL OR `AI Adoption Rate (%%)` = '' THEN 1 ELSE 0 END) AS missing_adoption_rate
FROM ai;
"""
pd.read_sql(query, engine)

missing_country  missing_year  missing_industry  missing_top_ai_tool_used  \
0              0.0           0.0               0.0                       0.0   

   missing_adoption_rate  
0                    0.0

<div style="background-color:#f5f0ff; padding:14px; border-left:6px solid #7e57c2; font-family:'Segoe UI', sans-serif; font-size:15px; line-height:1.6;">
  <i>All key fields are fully populated — no missing values in Country, Year, Industry, Top AI Tool, or AI Adoption Rate.</i>
</div>

<div style="background-color:#e6f4ea; padding:12px; border-left:5px solid #2e7d32; font-family:'Segoe UI', sans-serif;">
  <h4>🔹Handling Duplicates:
</div>

In [9]:
query = """
-- Identify duplicate rows using a window function
WITH NumberedRows AS (
    SELECT
        *,
        ROW_NUMBER() OVER(PARTITION BY
            Country,
            Year,
            Industry,
            `Top AI Tools Used`,
            `AI Adoption Rate (%%)`
        ORDER BY Year) AS row_num
    FROM ai
)
SELECT * FROM NumberedRows WHERE row_num > 1;
"""
pd.read_sql(query, engine)

Empty DataFrame
Columns: [Country, Year, Industry, AI Adoption Rate (%), AI-Generated Content Volume (TBs per year), Job Loss Due to AI (%), Revenue Increase Due to AI (%), Human-AI Collaboration Rate (%), Top AI Tools Used, Regulation Status, Consumer Trust in AI (%), Market Share of AI Companies (%), row_num]
Index: []

<div style="background-color:#f5f0ff; padding:14px; border-left:6px solid #7e57c2; font-family:'Segoe UI', sans-serif; font-size:15px; line-height:1.6;">
  <i>Dataset has no duplicate rows based on key attributes — all entries are unique.</i>
</div>

<div style="background-color:#e6f4ea; padding:12px; border-left:5px solid #2e7d32; font-family:'Segoe UI', sans-serif;">
  <h4>🔹Checking Datatypes:
</div>

In [10]:
query = """
SELECT 
    COLUMN_NAME,
    DATA_TYPE
FROM INFORMATION_SCHEMA.COLUMNS
WHERE TABLE_NAME = 'ai';
"""
pd.read_sql(query, engine)


COLUMN_NAME DATA_TYPE
0                         AI Adoption Rate (%)    double
1   AI-Generated Content Volume (TBs per year)    double
2                     Consumer Trust in AI (%)    double
3                                      Country      text
4              Human-AI Collaboration Rate (%)    double
5                                     Industry      text
6                       Job Loss Due to AI (%)    double
7             Market Share of AI Companies (%)    double
8                            Regulation Status      text
9               Revenue Increase Due to AI (%)    double
10                           Top AI Tools Used      text
11                                        Year       int

<div style="background-color:#e6f4ea; padding:12px; border-left:5px solid #2e7d32; font-family:'Segoe UI', sans-serif;">
  <h4>🔹Validating Data (Sanity Checks):
</div>

In [11]:
query = """
SELECT * FROM ai
WHERE
    `AI Adoption Rate (%%)` < 0 OR `AI Adoption Rate (%%)` > 100
    OR `Job Loss Due to AI (%%)` < 0 OR `Job Loss Due to AI (%%)` > 100
    OR `AI-Generated Content Volume (TBs per year)` < 0 OR `AI-Generated Content Volume (TBs per year)` > 100
    OR `Revenue Increase Due to AI (%%)` < 0 OR `Revenue Increase Due to AI (%%)` > 100
    OR `Human-AI Collaboration Rate (%%)` < 0 OR `Human-AI Collaboration Rate (%%)` > 100
    OR `Consumer Trust in AI (%%)` < 0 OR `Consumer Trust in AI (%%)` > 100
    OR `Market Share of AI Companies (%%)` < 0 OR `Market Share of AI Companies (%%)` > 100;
"""
pd.read_sql(query, engine)

Empty DataFrame
Columns: [Country, Year, Industry, AI Adoption Rate (%), AI-Generated Content Volume (TBs per year), Job Loss Due to AI (%), Revenue Increase Due to AI (%), Human-AI Collaboration Rate (%), Top AI Tools Used, Regulation Status, Consumer Trust in AI (%), Market Share of AI Companies (%)]
Index: []


<div style="background-color:#f5f0ff; padding:14px; border-left:6px solid #7e57c2; font-family:'Segoe UI', sans-serif; font-size:15px; line-height:1.6;">
  <i>Sanity Check Passed:</b> All key columns contain valid and consistent values — no anomalies detected.</i>
</div>


<div style="background-color:#e6f4ea; padding:12px; border-left:5px solid #2e7d32; font-family:'Segoe UI', sans-serif;">
  <h4> 🔹 Cross-Column Sanity Check : </h4>
   - AI Adoption vs Other Metrics -- AI Adoption = 0 but Other Metrics Are > 0<br>
   - Why it's a problem: If no AI is adopted, we shouldn't see AI-driven effects like job loss, revenue gains, collaboration, market growth, consumer trust or ai-generated content.
</div>

In [12]:
query = """
SELECT * FROM ai
WHERE `AI Adoption Rate (%%)` = 0 AND (
      `Job Loss Due to AI (%%)` > 0 OR
      `Revenue Increase Due to AI (%%)` > 0 OR
      `Human-AI Collaboration Rate (%%)` > 0 OR
      `Market Share of AI Companies (%%)` > 0 OR
      `Consumer Trust in AI (%%)` > 0 OR
      `AI-Generated Content Volume (TBs per year)` > 0)
"""
pd.read_sql(query, engine)

Empty DataFrame
Columns: [Country, Year, Industry, AI Adoption Rate (%), AI-Generated Content Volume (TBs per year), Job Loss Due to AI (%), Revenue Increase Due to AI (%), Human-AI Collaboration Rate (%), Top AI Tools Used, Regulation Status, Consumer Trust in AI (%), Market Share of AI Companies (%)]
Index: []

<div style="background-color:#f5f0ff; padding:14px; border-left:6px solid #7e57c2; font-family:'Segoe UI', sans-serif; font-size:15px; line-height:1.6;">
  <i>Cross-column sanity check passed — No conflicting records where AI adoption is 0 but other metrics are > 0.</i>
</div>

<!-- End of Previous Section -->
<hr style="border: none; border-top: 2px dashed #ccc; margin: 30px 0;" />

<div style="background-color:#ffe5d9; padding:10px; border-left:6px solid #ff7043; font-family:'Segoe UI', sans-serif;">
  <h2 style="margin:0; color:#d84315;">💡 AI-Driven Insights</h2>
</div>

<p style="font-family:'Segoe UI', sans-serif; font-size:15px; line-height:1.6;">
  This section highlights meaningful patterns, business opportunities, and global trends uncovered through advanced SQL queries and analytical reasoning.
</p>

<!-- Title + Objective Section -->
<div style="font-family:'Segoe UI', sans-serif; margin: 20px 0;">

  <!-- Title Banner -->
  <div style="background: linear-gradient(to right, #7b1fa2, #ba68c8); padding: 12px 20px; border-radius: 8px;">
    <h2 style="color:white; margin:0;">💥 AI Impact Power Index by Country</h2>
  </div>

  <!-- Objective Block -->
  <div style="background-color:#f3e5f5; padding: 14px 20px; border-left: 5px solid #8e24aa; border-radius: 0 0 8px 8px;">
    <p style="margin:0;"><b>🎯 Objective:</b> Combines multiple KPIs — AI Adoption, Revenue Gain, Collaboration Rate, and Market Share — into a single composite score to rank each country's overall AI impact.</p>
  </div>
</div>


In [13]:
query = """
WITH ImpactData AS (
  SELECT
    Country,
    ROUND(AVG(`AI Adoption Rate (%%)`), 2) AS `Avg AI Adoption Rate (%%)`,
    ROUND(AVG(`Revenue Increase Due to AI (%%)`), 2) AS `Avg Revenue Increase Due to AI (%%)`,
    ROUND(AVG(`Human-AI Collaboration Rate (%%)`), 2) AS `Avg Human-AI Collaboration Rate (%%)`,
    ROUND(AVG(`Market Share of AI Companies (%%)`), 2) AS `Avg Market Share of AI Companies (%%)`,
    ROUND((
      AVG(`AI Adoption Rate (%%)`) + AVG(`Revenue Increase Due to AI (%%)`) +
      AVG(`Human-AI Collaboration Rate (%%)`) + AVG(`Market Share of AI Companies (%%)`)
    ) / 4, 2) AS AI_Impact_Index
  FROM ai
  GROUP BY Country
),
RankedImpact AS (
  SELECT *,
         RANK() OVER (ORDER BY AI_Impact_Index DESC) AS `AI_Rank`
  FROM ImpactData
)
SELECT *
FROM RankedImpact;
"""
pd.read_sql(query, engine)


Country  Avg AI Adoption Rate (%)  Avg Revenue Increase Due to AI (%)  \
0    Australia                     56.08                               49.70   
1           UK                     64.69                               36.12   
2        China                     52.89                               42.60   
3       France                     56.52                               39.55   
4        India                     51.81                               40.08   
5       Canada                     52.19                               38.75   
6          USA                     52.08                               43.52   
7      Germany                     51.46                               40.58   
8  South Korea                     50.56                               34.25   
9        Japan                     54.21                               34.61   

   Avg Human-AI Collaboration Rate (%)  Avg Market Share of AI Companies (%)  \
0                                59.48                                 23.69   
1                                58.27                                 25.14   
2                                54.51                                 27.06   
3                                52.76                                 27.19   
4                                54.20                                 28.34   
5                                55.20                                 28.28   
6                                57.26                                 20.98   
7                                49.63                                 30.46   
8                                59.47                                 27.33   
9                                42.38                                 27.15   

   AI_Impact_Index  AI_Rank  
0            47.24        1  
1            46.06        2  
2            44.26        3  
3            44.01        4  
4            43.61        5  
5            43.60        6  
6            43.46        7  
7            43.03        8  
8            42.90        9  
9            39.59       10

<!-- Separated Insight Card -->
<div style="font-family:'Segoe UI', sans-serif; background-color:#ede7f6; padding: 14px 20px; border-left: 5px solid #6a1b9a; border-radius: 8px; margin-bottom: 30px;">
  <b>💡 Insight:</b> Australia leads the AI Impact Index, showing that strong human-AI collaboration and balanced adoption drive greater overall AI effectiveness than market share alone.
</div>

<hr style="border: none; border-top: 2px dashed #ccc; margin: 30px 0;" />


<!-- Title + Objective -->
<div style="font-family:'Segoe UI', sans-serif; margin: 20px 0;">

  <!-- Title Banner -->
  <div style="background: linear-gradient(to right, #1565c0, #42a5f5); padding: 12px 20px; border-radius: 8px;">
    <h2 style="color:white; margin:0;">⚠️ Identify Industries with High AI Adoption but Low Economic Return</h2>
  </div>

  <!-- Objective Block -->
  <div style="background-color:#e3f2fd; padding: 14px 20px; border-left: 5px solid #1976d2; border-radius: 0 0 8px 8px;">
    <p style="margin:0;"><b>🎯 Objective:</b> Highlights industries where high AI adoption isn't yielding proportionate economic returns, helping stakeholders identify 
efficiency gaps or implementation issues.</p>
  </div>
</div>

In [14]:
query = """
WITH IndustryStats AS (
    SELECT
        Industry,
        Country,
        ROUND(AVG(`AI Adoption Rate (%%)`), 2) AS Avg_Adoption,
        ROUND(AVG(`Revenue Increase Due to AI (%%)`), 2) AS Avg_Revenue
    FROM ai
    GROUP BY Industry, Country
),
RankedIndustries AS (
    SELECT *,
           (Avg_Adoption - Avg_Revenue) AS Diff_Avg_Adoption_and_Avg_Revenue
    FROM IndustryStats
    ORDER BY Diff_Avg_Adoption_and_Avg_Revenue DESC
)
SELECT
    Industry,
    Country,
    Avg_Adoption,
    Avg_Revenue,
    Diff_Avg_Adoption_and_Avg_Revenue
FROM RankedIndustries
LIMIT 10;
"""
pd.read_sql(query, engine)

Industry      Country  Avg_Adoption  Avg_Revenue  \
0     Automotive      Germany         89.44        12.14   
1        Finance           UK         88.23        10.99   
2         Gaming  South Korea         78.26         6.73   
3     Automotive        Japan         92.96        25.17   
4         Retail        China         82.41        17.78   
5        Finance      Germany         79.93        15.92   
6  Manufacturing       France         93.16        35.31   
7     Healthcare       France         77.33        23.16   
8          Legal           UK         50.34         0.58   
9      Marketing       Canada         59.88        10.98   

   Diff_Avg_Adoption_and_Avg_Revenue  
0                              77.30  
1                              77.24  
2                              71.53  
3                              67.79  
4                              64.63  
5                              64.01  
6                              57.85  
7                              54.17  
8                              49.76  
9                              48.90

<!-- Insight Block (Blue Theme) -->
<div style="font-family:'Segoe UI', sans-serif; background-color:#e3f2fd; padding: 14px 20px; border-left: 5px solid #1976d2; border-radius: 8px; margin-bottom: 30px;">
  <b>💡 Insight:</b> Despite strong AI adoption, industries like <b>Automotive in Germany</b> and <b>Finance in the UK</b> show significant gaps in economic return. This indicates that high investment in AI alone isn’t enough — strategic deployment and alignment with business goals are essential to drive real value.
</div>


<hr style="border: none; border-top: 2px dashed #ccc; margin: 30px 0;" />


<!-- Title + Objective -->
<div style="font-family:'Segoe UI', sans-serif; margin: 20px 0;">

  <!-- Title Banner -->
  <div style="background: linear-gradient(to right, #e65100, #ffb74d); padding: 12px 20px; border-radius: 8px;">
    <h2 style="color:white; margin:0;">🔁 Rank the Most Volatile Industry-Country Pairs in AI Adoption</h2>
  </div>

  <!-- Objective Block -->
  <div style="background-color:#fff3e0; padding: 14px 20px; border-left: 5px solid #fb8c00; border-radius: 0 0 8px 8px;">
    <p style="margin:0;"><b>🎯 Objective:</b> Reveal industry-country pairs with the highest fluctuations in AI adoption over time by ranking them using the standard deviation of their adoption rates — helping stakeholders flag inconsistent implementation.</p>
  </div>
</div>

In [15]:
query = """
WITH IndustryStats AS (
    SELECT
        Industry,
        Country,
        COUNT(*) AS diff_number_entries,
        ROUND(AVG(`AI Adoption Rate (%%)`), 2) AS Avg_Adoption,
        ROUND(STDDEV_POP(`AI Adoption Rate (%%)`), 2) AS StdDev_Adoption
    FROM ai
    GROUP BY Industry, Country
    HAVING diff_number_entries >= 5
),
Ranked AS (
    SELECT *
    FROM IndustryStats
    ORDER BY StdDev_Adoption DESC
)
SELECT * FROM Ranked;
"""
pd.read_sql(query, engine)


Industry      Country  diff_number_entries  Avg_Adoption  \
0         Gaming           UK                    6         67.56   
1     Automotive           UK                    5         56.97   
2     Healthcare  South Korea                    5         52.59   
3  Manufacturing        Japan                    5         64.57   
4         Retail        India                    5         60.84   
5          Media       France                    7         51.35   
6          Media        Japan                    5         45.96   
7         Retail          USA                    5         33.96   

   StdDev_Adoption  
0            27.47  
1            26.05  
2            25.43  
3            24.88  
4            24.21  
5            23.94  
6            23.29  
7            18.48

<!-- Insight Block -->
<div style="font-family:'Segoe UI', sans-serif; background-color:#fff8e1; padding: 14px 20px; border-left: 5px solid #f57c00; border-radius: 8px; margin-bottom: 30px;">
  <b>💡 Insight:</b> <b>Gaming in the UK</b> shows the highest volatility in AI adoption (StdDev: 27.47), pointing to unpredictable implementation — possibly driven by fluctuating investment, innovation cycles, or inconsistent policies.
</div>

<hr style="border: none; border-top: 2px dashed #ccc; margin: 30px 0;" />


<!-- Title + Objective -->
<div style="font-family:'Segoe UI', sans-serif; margin: 20px 0;">

  <!-- Title Banner -->
  <div style="background: linear-gradient(to right, #2e7d32, #81c784); padding: 12px 20px; border-radius: 8px;">
    <h2 style="color:white; margin:0;">📈 Identify Early Adopters (Industries Above Global Average AI Adoption in 2020)</h2>
  </div>

  <!-- Objective Block -->
  <div style="background-color:#e8f5e9; padding: 14px 20px; border-left: 5px solid #43a047; border-radius: 0 0 8px 8px;">
    <p style="margin:0;"><b>🎯 Objective:</b> Highlight industries and countries that were ahead of the global curve in AI adoption during 2020, signaling early leadership, innovation readiness, and proactive digital transformation.</p>
  </div>
</div>

In [16]:
query = """
WITH Global2020Stats AS (
    SELECT 
        ROUND(AVG(`AI Adoption Rate (%%)`), 2) AS Global_Avg_2020
    FROM ai
    WHERE Year = 2020
),
Industry2020 AS (
    SELECT 
        Country,
        Industry,
        `AI Adoption Rate (%%)`,
        RANK() OVER (PARTITION BY Industry ORDER BY `AI Adoption Rate (%%)` DESC) AS Adoption_Rank
    FROM ai
    WHERE Year = 2020
),
FilteredIndustries AS (
    SELECT 
        i.Country,
        i.Industry,
        i.`AI Adoption Rate (%%)`,
        g.Global_Avg_2020,
        i.Adoption_Rank
    FROM Industry2020 i
    JOIN Global2020Stats g ON 1=1
    WHERE i.`AI Adoption Rate (%%)` > g.Global_Avg_2020
)
SELECT *
FROM FilteredIndustries
ORDER BY `AI Adoption Rate (%%)` DESC
LIMIT 10;
"""
pd.read_sql(query, engine)


Country       Industry  AI Adoption Rate (%)  Global_Avg_2020  \
0    India         Retail                 93.72            50.99   
1    China  Manufacturing                 93.49            50.99   
2   France  Manufacturing                 93.16            50.99   
3       UK        Finance                 88.23            50.99   
4    China         Retail                 82.41            50.99   
5    India     Automotive                 80.04            50.99   
6   France     Automotive                 79.59            50.99   
7    China  Manufacturing                 72.88            50.99   
8  Germany        Finance                 71.90            50.99   
9   France      Marketing                 71.00            50.99   

   Adoption_Rank  
0              1  
1              1  
2              2  
3              1  
4              2  
5              1  
6              2  
7              3  
8              2  
9              1

<!-- Insight Block -->
<div style="font-family:'Segoe UI', sans-serif; background-color:#f1f8e9; padding: 14px 20px; border-left: 5px solid #558b2f; border-radius: 8px; margin-bottom: 30px;">
  <b>💡 Insight:</b> <b>Retail in India (93.72%)</b> and <b>Manufacturing in China (93.49%)</b> were standout early adopters in 2020, demonstrating strategic foresight and a competitive edge in embracing AI transformation ahead of peers.
</div>

<hr style="border: none; border-top: 2px dashed #ccc; margin: 30px 0;" />


<!-- Title + Objective -->
<div style="font-family:'Segoe UI', sans-serif; margin: 20px 0;">

  <!-- Title Banner -->
  <div style="background: linear-gradient(to right, #f9a825, #ffeb3b); padding: 12px 20px; border-radius: 8px;">
    <h2 style="color:#212121; margin:0;">🏆 Year-over-Year Top Performing Country by Economic Gain per AI Unit</h2>
  </div>

  <!-- Objective Block -->
  <div style="background-color:#fff8e1; padding: 14px 20px; border-left: 5px solid #fbc02d; border-radius: 0 0 8px 8px;">
    <p style="margin:0;"><b>🎯 Objective:</b> Identify the most efficient countries each year in monetizing AI, by ranking them based on revenue gained per unit of AI adoption — a powerful metric of AI investment performance.</p>
  </div>
</div>

In [17]:
query = """
SELECT *
FROM (
  SELECT
    Year,
    Country,
    ROUND(AVG(`AI Adoption Rate (%%)`), 2) AS Avg_Adoption,
    ROUND(AVG(`Revenue Increase Due to AI (%%)`), 2) AS Avg_Revenue,
    ROUND(AVG(`Revenue Increase Due to AI (%%)`) / NULLIF(AVG(`AI Adoption Rate (%%)`), 0), 2) AS Economic_Gain_Per_AI_Unit,
    RANK() OVER (PARTITION BY Year ORDER BY AVG(`Revenue Increase Due to AI (%%)`) / NULLIF(AVG(`AI Adoption Rate (%%)`), 0) DESC) AS Performance_Rank
  FROM ai
  GROUP BY Year, Country
  HAVING AVG(`AI Adoption Rate (%%)`) > 0
) sub
WHERE Performance_Rank = 1
ORDER BY Year;
"""
pd.read_sql(query, engine)


Year      Country  Avg_Adoption  Avg_Revenue  Economic_Gain_Per_AI_Unit  \
0  2020  South Korea         23.29        49.24                       2.11   
1  2021       Canada         49.72        66.78                       1.34   
2  2022        China         37.49        44.73                       1.19   
3  2023          USA         47.57        48.26                       1.01   
4  2024      Germany         17.05        70.85                       4.16   
5  2025    Australia         49.24        54.54                       1.11   

   Performance_Rank  
0                 1  
1                 1  
2                 1  
3                 1  
4                 1  
5                 1

<!-- Insight Block -->
<div style="font-family:'Segoe UI', sans-serif; background-color:#fff3e0; padding: 14px 20px; border-left: 5px solid #fb8c00; border-radius: 8px; margin-bottom: 30px;">
  <b>💡 Insight:</b> <b>Germany in 2024</b> led the global chart with an impressive Economic Gain per AI Unit of <b>4.16</b>, proving it as the most economically efficient nation in converting AI efforts into real revenue that year.
</div>

<hr style="border: none; border-top: 2px dashed #ccc; margin: 30px 0;" />


<!-- Title + Objective -->
<div style="font-family:'Segoe UI', sans-serif; margin: 20px 0;">

  <!-- Title Banner -->
  <div style="background: linear-gradient(to right, #00695c, #4db6ac); padding: 12px 20px; border-radius: 8px;">
    <h2 style="color:white; margin:0;">🧠 AI Maturity vs Market Presence</h2>
  </div>

  <!-- Objective Block -->
  <div style="background-color:#e0f2f1; padding: 14px 20px; border-left: 5px solid #00897b; border-radius: 0 0 8px 8px;">
    <p style="margin:0;"><b>🎯 Objective:</b> Identify industry-country combinations with high <b>Human-AI collaboration</b> and <b>consumer trust</b>, indicating mature AI integration, and analyze how these align with AI companies’ <b>market share</b>.</p>
  </div>
</div>


In [18]:
query = """
WITH CollabTrust AS (
  SELECT 
    Country,
    Industry,
    ROUND(AVG(`Human-AI Collaboration Rate (%%)`), 2) AS Avg_Collab,
    ROUND(AVG(`Consumer Trust in AI (%%)`), 2) AS Avg_Trust,
    ROUND(AVG(`Market Share of AI Companies (%%)`), 2) AS Avg_Market_Share,
    COUNT(*) AS Total_Count
  FROM ai
  GROUP BY Country, Industry
),
MaturityScored AS (
  SELECT *,
    ROUND((Avg_Collab + Avg_Trust) / 2, 2) AS AI_Maturity_Index,
    CASE 
      WHEN Avg_Trust >= 75 THEN 'High Trust'
      WHEN Avg_Trust >= 50 THEN 'Moderate Trust'
      ELSE 'Low Trust'
    END AS Trust_Tier
  FROM CollabTrust
)
SELECT *
FROM MaturityScored
WHERE AI_Maturity_Index >= 60 AND Total_Count >= 3
ORDER BY AI_Maturity_Index DESC;
"""
pd.read_sql(query, engine)


Country    Industry  Avg_Collab  Avg_Trust  Avg_Market_Share  \
0        India      Retail       72.66      68.30             26.65   
1       Canada      Gaming       71.58      67.21             29.18   
2    Australia  Automotive       68.75      65.03             31.60   
3  South Korea       Legal       60.87      71.52             17.71   
4        Japan   Marketing       65.37      58.18             17.86   
5      Germany       Media       45.90      75.30             32.18   

   Total_Count  AI_Maturity_Index      Trust_Tier  
0            5              70.48  Moderate Trust  
1            3              69.40  Moderate Trust  
2            3              66.89  Moderate Trust  
3            3              66.19  Moderate Trust  
4            3              61.78  Moderate Trust  
5            4              60.60      High Trust

<!-- Insight Block -->
<div style="font-family:'Segoe UI', sans-serif; background-color:#f1f8e9; padding: 14px 20px; border-left: 5px solid #689f38; border-radius: 8px; margin-bottom: 30px;">
  <b>💡 Insight:</b> <b>India’s Retail sector</b> stands out with a <b>Human-AI Collaboration Rate of 72.66%</b> and <b>Consumer Trust of 68.30%</b>, forming an <b>AI Maturity Index of 70.48</b>.  
  Despite being in the "Moderate Trust" tier, it showcases strong integration with a healthy <b>AI Market Share of 26.65%</b>, highlighting India's effective and trusted AI implementation in Retail.
</div>


<hr style="border: none; border-top: 2px dashed #ccc; margin: 30px 0;" />


<!-- Title + Objective -->
<div style="font-family:'Segoe UI', sans-serif; margin: 20px 0;">

  <!-- Title Banner -->
  <div style="background: linear-gradient(to right, #283593, #ffb74d); padding: 12px 20px; border-radius: 8px;">
    <h2 style="color:white; margin:0;">🔐 AI Content Creation vs Economic and Workforce Impact</h2>
  </div>

  <!-- Objective Block -->
  <div style="background-color:#e8eaf6; padding: 14px 20px; border-left: 5px solid #3f51b5; border-radius: 0 0 8px 8px;">
    <p style="margin:0;"><b>🎯 Objective:</b> Identify top country-industry pairs in AI-generated content output and examine how this correlates with their <b>economic returns</b> and <b>workforce impact</b>, shedding light on the uneven distribution of AI’s benefits and disruptions.</p>
  </div>
</div>


In [19]:
query = """
WITH CountryIndustryStats AS (
    SELECT 
        Country,
        Industry,
        ROUND(AVG(`AI-Generated Content Volume (TBs per year)`), 2) AS Avg_Content_TB,
        ROUND(AVG(`Revenue Increase Due to AI (%%)`), 2) AS Avg_Revenue,
        ROUND(AVG(`Job Loss Due to AI (%%)`), 2) AS Avg_Job_Loss
    FROM ai
    GROUP BY Country, Industry
),
RankedImpact AS (
    SELECT *,
        RANK() OVER (ORDER BY Avg_Content_TB DESC) AS Content_Rank,
        RANK() OVER (ORDER BY Avg_Revenue DESC) AS Revenue_Rank,
        RANK() OVER (ORDER BY Avg_Job_Loss DESC) AS JobLoss_Rank
    FROM CountryIndustryStats
)
SELECT *
FROM RankedImpact
ORDER BY Content_Rank
LIMIT 10;
"""
pd.read_sql(query, engine)


Country       Industry  Avg_Content_TB  Avg_Revenue  Avg_Job_Loss  \
0          USA     Automotive           96.13        45.60         10.66   
1        Japan          Legal           95.44        34.13         23.10   
2        China      Education           94.67        44.70         37.59   
3       France        Finance           94.47        28.86         26.42   
4        India          Legal           92.91        14.73         42.89   
5  South Korea         Retail           85.80        38.83         41.38   
6        China     Automotive           85.74        75.69         15.77   
7  South Korea     Automotive           82.97        60.61         37.08   
8        China         Retail           77.29        17.78         11.75   
9       Canada  Manufacturing           73.60        66.78         44.52   

   Content_Rank  Revenue_Rank  JobLoss_Rank  
0             1            31            76  
1             2            55            50  
2             3            34            14  
3             4            63            41  
4             5            77             7  
5             6            45             8  
6             7             3            73  
7             8             9            15  
8             9            73            75  
9            10             6             5

<!-- Insight Block -->
<div style="font-family:'Segoe UI', sans-serif; background-color:#fff8e1; padding: 14px 20px; border-left: 5px solid #f57c00; border-radius: 8px; margin-bottom: 30px;">
  <b>💡 Insight:</b> The <b>USA–Automotive</b> sector tops AI content creation (96.13 TB/year) but shows minimal job displacement (76th in job loss), while <b>India–Legal</b> ranks 5th in content generation yet faces the 7th highest job loss.  
  This reveals that <b>high AI output doesn’t guarantee high revenue</b> and may often correlate with <b>greater workforce disruption</b>, underlining the need for balanced AI strategy that weighs both economic gains and labor implications.
</div>


<hr style="border: none; border-top: 2px dashed #ccc; margin: 30px 0;" />


<!-- Title + Objective -->
<div style="font-family:'Segoe UI', sans-serif; margin: 20px 0;">

  <!-- Title Banner -->
  <div style="background: linear-gradient(to right, #00695c, #4db6ac); padding: 12px 20px; border-radius: 8px;">
    <h2 style="color:white; margin:0;">🧮 Identifying Significant Year-over-Year (YoY) Surges in AI Adoption</h2>
  </div>

  <!-- Objective Block -->
  <div style="background-color:#e0f2f1; padding: 14px 20px; border-left: 5px solid #00796b; border-radius: 0 0 8px 8px;">
    <p style="margin:0;"><b>🎯 Objective:</b> Identify country-industry pairs that saw major leaps in AI adoption from one year to the next, using absolute and percentage growth to spotlight breakout moments of digital acceleration.</p>
  </div>
</div>

In [20]:
query = """
WITH YearlyAvg AS (
  SELECT
    Country,
    Industry,
    Year,
    ROUND(AVG(`AI Adoption Rate (%%)`), 2) AS Avg_Adoption
  FROM ai
  GROUP BY Country, Industry, Year
),
AdoptionDiff AS (
  SELECT
    Country,
    Industry,
    Year,
    Avg_Adoption,
    LAG(Avg_Adoption) OVER (PARTITION BY Country, Industry ORDER BY Year) AS Prev_Year_Adoption
  FROM YearlyAvg
),
YoYSpikes AS (
  SELECT
    Country,
    Industry,
    Year,
    Avg_Adoption,
    Prev_Year_Adoption,
    ROUND((Avg_Adoption - Prev_Year_Adoption), 2) AS Absolute_Growth,
    ROUND(
      CASE 
        WHEN Prev_Year_Adoption > 0 THEN 
            ((Avg_Adoption - Prev_Year_Adoption) * 100.0) / Prev_Year_Adoption
        ELSE NULL
      END, 2
    ) AS Percentage_Growth,
    CASE 
      WHEN (Avg_Adoption - Prev_Year_Adoption) >= 75 THEN 'Extreme Surge'
      WHEN (Avg_Adoption - Prev_Year_Adoption) >= 50 THEN 'High Growth'
      WHEN (Avg_Adoption - Prev_Year_Adoption) >= 30 THEN 'Moderate Growth'
      WHEN (Avg_Adoption - Prev_Year_Adoption) >= 10 THEN 'Mild Growth'
      ELSE 'Stable or Decline'
    END AS Growth_Label,
    CASE
      WHEN Prev_Year_Adoption IS NULL THEN 'Not Applicable'
      WHEN ((Avg_Adoption - Prev_Year_Adoption) * 100.0) / Prev_Year_Adoption > 200 THEN '⚠ >200%% Surge'
      WHEN ((Avg_Adoption - Prev_Year_Adoption) * 100.0) / Prev_Year_Adoption > 150 THEN '🚀 150-200%%'
      WHEN ((Avg_Adoption - Prev_Year_Adoption) * 100.0) / Prev_Year_Adoption > 100 THEN '🔥 100-150%%'
      WHEN ((Avg_Adoption - Prev_Year_Adoption) * 100.0) / Prev_Year_Adoption > 75 THEN '⬆ 75-100%%'
      WHEN ((Avg_Adoption - Prev_Year_Adoption) * 100.0) / Prev_Year_Adoption > 50 THEN '⬆ 50-75%%'
      WHEN ((Avg_Adoption - Prev_Year_Adoption) * 100.0) / Prev_Year_Adoption > 25 THEN '↗ 25-50%%'
      ELSE '↘ <25%%'
    END AS Percent_Growth_Bucket
  FROM AdoptionDiff
  WHERE Prev_Year_Adoption IS NOT NULL
)
SELECT *
FROM YoYSpikes
WHERE Absolute_Growth > 40
ORDER BY Absolute_Growth DESC
LIMIT 10;
"""
pd.read_sql(query, engine)


Country    Industry  Year  Avg_Adoption  Prev_Year_Adoption  \
0  South Korea     Finance  2023         83.74               18.91   
1       Canada      Gaming  2025         94.76               31.96   
2           UK   Education  2023         82.82               21.57   
3       France       Media  2025         79.64               23.27   
4        China      Gaming  2025         70.04               13.90   
5       Canada       Legal  2025         82.25               27.53   
6  South Korea  Healthcare  2021         60.86               10.53   
7          USA     Finance  2023         83.20               36.52   
8        China       Media  2021         85.09               39.22   
9        Japan   Marketing  2024         88.16               42.72   

   Absolute_Growth  Percentage_Growth     Growth_Label Percent_Growth_Bucket  
0            64.83             342.83      High Growth         ⚠ >200% Surge  
1            62.80             196.50      High Growth            🚀 150-200%  
2            61.25             283.96      High Growth         ⚠ >200% Surge  
3            56.37             242.24      High Growth         ⚠ >200% Surge  
4            56.14             403.88      High Growth         ⚠ >200% Surge  
5            54.72             198.76      High Growth            🚀 150-200%  
6            50.33             477.97      High Growth         ⚠ >200% Surge  
7            46.68             127.82  Moderate Growth            🔥 100-150%  
8            45.87             116.96  Moderate Growth            🔥 100-150%  
9            45.44             106.37  Moderate Growth            🔥 100-150%

<!-- Insight Block -->
<div style="font-family:'Segoe UI', sans-serif; background-color:#e0f7fa; padding: 14px 20px; border-left: 5px solid #00838f; border-radius: 8px; margin-bottom: 30px;">
  <b>💡 Insight:</b> <b>South Korea’s Finance (2023)</b> and <b>Canada’s Gaming (2025)</b> sectors experienced exceptional surges in AI adoption — with growth over <b>60 points</b> — reflecting aggressive AI investments and transformative strategy execution in these industries.<br>"We use percentage growth to capture relative change. If a country/industry started with low adoption and made a significant leap, % growth naturally exceeds 100%. That’s expected behavior and actually highlights rapid transformation or aggressive AI investments."
</div>


<hr style="border: none; border-top: 2px dashed #ccc; margin: 30px 0;" />


<!-- Title + Objective -->
<div style="font-family:'Segoe UI', sans-serif; margin: 20px 0;">

  <!-- Title Banner -->
  <div style="background: linear-gradient(to right, #b71c1c, #f06292); padding: 12px 20px; border-radius: 8px;">
    <h2 style="color:white; margin:0;">📉 AI Adoption and Job Loss Paradox: Industries with High Adoption but Decreasing Job Loss</h2>
  </div>

  <!-- Objective Block -->
  <div style="background-color:#fce4ec; padding: 14px 20px; border-left: 5px solid #d81b60; border-radius: 0 0 8px 8px;">
    <p style="margin:0;"><b>🎯 Objective:</b> Several industries show a paradox where AI adoption is high (>50%) yet job loss due to AI is decreasing year-over-year, suggesting effective AI integration without proportional workforce reduction.</p>
  </div>
</div>


In [21]:
query = """
WITH YearlyStats AS (
  SELECT
    Country,
    Industry,
    Year,
    ROUND(AVG(`AI Adoption Rate (%%)`), 2) AS Avg_Adoption,
    ROUND(AVG(`Job Loss Due to AI (%%)`), 2) AS Avg_Job_Loss
  FROM ai
  GROUP BY Country, Industry, Year
),
WithPrevYear AS (
  SELECT
    curr.Country,
    curr.Industry,
    curr.Year,
    curr.Avg_Adoption,
    curr.Avg_Job_Loss,
    prev.Avg_Job_Loss AS Prev_Job_Loss
  FROM YearlyStats curr
  LEFT JOIN YearlyStats prev
    ON curr.Country = prev.Country
    AND curr.Industry = prev.Industry
    AND curr.Year = prev.Year + 1
),
Paradox AS (
  SELECT *,
    ROUND(((Prev_Job_Loss - Avg_Job_Loss) * 100.0) / Prev_Job_Loss, 2) AS Job_Loss_Decline_Perc,
    CASE 
      WHEN Prev_Job_Loss IS NOT NULL AND Avg_Job_Loss < Prev_Job_Loss THEN 'Yes'
      ELSE 'No'
    END AS JobLoss_Decreasing
  FROM WithPrevYear
)
SELECT *
FROM Paradox
WHERE JobLoss_Decreasing = 'Yes' AND Avg_Adoption > 50
ORDER BY Year DESC, Avg_Adoption DESC
LIMIT 10;
"""
pd.read_sql(query, engine)


Country       Industry  Year  Avg_Adoption  Avg_Job_Loss  \
0       France          Media  2025         79.64         12.32   
1  South Korea          Legal  2025         52.86         17.30   
2        Japan      Marketing  2024         88.16          2.86   
3    Australia     Automotive  2024         78.21          6.54   
4           UK         Gaming  2024         77.52         14.58   
5        Japan  Manufacturing  2023         78.57         17.23   
6          USA         Gaming  2022         93.00         12.92   
7    Australia         Gaming  2022         89.96         25.11   
8           UK         Gaming  2022         59.96         14.26   
9        Japan          Media  2021         76.22         15.67   

   Prev_Job_Loss  Job_Loss_Decline_Perc JobLoss_Decreasing  
0          44.22                  72.14                Yes  
1          29.64                  41.63                Yes  
2          29.40                  90.27                Yes  
3          33.21                  80.31                Yes  
4          41.67                  65.01                Yes  
5          25.18                  31.57                Yes  
6          40.55                  68.14                Yes  
7          43.30                  42.01                Yes  
8          23.70                  39.83                Yes  
9          28.58                  45.17                Yes

<!-- Insight Block -->
<div style="font-family:'Segoe UI', sans-serif; background-color:#f3e5f5; padding: 14px 20px; border-left: 5px solid #ad1457; border-radius: 8px; margin-bottom: 30px;">
  <b>💡 Insight:</b> In industries like <b>Media in France</b> (79.6%) and <b>Marketing in Japan</b> (88.2%), AI adoption surged while job loss dropped by over 70–90%, showing a clear paradox — where technology boosts productivity without cutting jobs.
</div>


<hr style="border: none; border-top: 2px dashed #ccc; margin: 30px 0;" />


<!-- Title + Objective -->
<div style="font-family:'Segoe UI', sans-serif; margin: 20px 0;">

  <!-- Title Banner -->
  <div style="background: linear-gradient(to right, #2e7d32, #66bb6a); padding: 12px 20px; border-radius: 8px;">
    <h2 style="color:white; margin:0;">📊 AI Tool Transitions Driving Business Impact Across Industries</h2>
  </div>

  <!-- Objective Block -->
  <div style="background-color:#e8f5e9; padding: 14px 20px; border-left: 5px solid #388e3c; border-radius: 0 0 8px 8px;">
    <p style="margin:0;"><b>🎯 Objective:</b> Identify industry-country cases where changes in AI tools led to significant increases in revenue and collaboration, highlighting which tool shifts were most impactful.</p>
  </div>

</div>


In [22]:
query = """
WITH ToolChangePairs AS (
  SELECT
    a.Country,
    a.Industry,
    a.Year AS Earlier_Year,
    b.Year AS Later_Year,
    a.`Top AI Tools Used` AS Earlier_Tool,
    b.`Top AI Tools Used` AS Later_Tool,
    b.`Revenue Increase Due to AI (%%)` - a.`Revenue Increase Due to AI (%%)` AS Revenue_Change,
    b.`Human-AI Collaboration Rate (%%)` - a.`Human-AI Collaboration Rate (%%)` AS Collab_Change,
    ABS(b.Year - a.Year) AS Year_Gap
  FROM ai a
  JOIN ai b
    ON a.Country = b.Country
    AND a.Industry = b.Industry
    AND b.Year > a.Year
    AND a.`Top AI Tools Used` <> b.`Top AI Tools Used`
    AND ABS(b.Year - a.Year) <= 2
),
RankedImpact AS (
  SELECT *,
         ROUND((Revenue_Change + Collab_Change)/2, 2) AS Avg_Impact_Score,
         CASE 
           WHEN ROUND((Revenue_Change + Collab_Change)/2, 2) >= 20 THEN 'Major Impact'
           WHEN ROUND((Revenue_Change + Collab_Change)/2, 2) >= 10 THEN 'Moderate Impact'
           ELSE 'Low Impact'
         END AS Impact_Tier
  FROM ToolChangePairs
)
SELECT *
FROM RankedImpact
WHERE Avg_Impact_Score > 10
ORDER BY Avg_Impact_Score DESC
LIMIT 10;
"""
pd.read_sql(query, engine)


Country   Industry  Earlier_Year  Later_Year      Earlier_Tool  \
0      Germany  Education          2021        2022              Bard   
1           UK     Gaming          2022        2024        Midjourney   
2       France      Media          2023        2024         Synthesia   
3       France      Media          2022        2024         Synthesia   
4       France      Media          2023        2025         Synthesia   
5           UK     Gaming          2022        2023        Midjourney   
6           UK     Gaming          2022        2024        Midjourney   
7  South Korea      Legal          2023        2024  Stable Diffusion   
8        India  Education          2020        2021  Stable Diffusion   
9    Australia     Gaming          2021        2022            DALL-E   

         Later_Tool  Revenue_Change  Collab_Change  Year_Gap  \
0  Stable Diffusion           73.54          34.99         1   
1            DALL-E           64.10          32.79         2   
2           ChatGPT           54.10          41.89         1   
3           ChatGPT           50.33          43.16         2   
4            DALL-E           40.97          46.24         2   
5           ChatGPT           52.46          26.49         1   
6            DALL-E           53.14          16.56         2   
7            DALL-E           31.62          27.21         1   
8            DALL-E           35.65          17.59         1   
9            Claude           63.41         -10.21         1   

   Avg_Impact_Score   Impact_Tier  
0             54.26  Major Impact  
1             48.45  Major Impact  
2             47.99  Major Impact  
3             46.74  Major Impact  
4             43.60  Major Impact  
5             39.48  Major Impact  
6             34.85  Major Impact  
7             29.42  Major Impact  
8             26.62  Major Impact  
9             26.60  Major Impact

<!-- Insight Block -->
<div style="font-family:'Segoe UI', sans-serif; background-color:#f1f8e9; padding: 14px 20px; border-left: 5px solid #558b2f; border-radius: 8px; margin-bottom: 30px;">
  <b>💡 Insight:</b> Transitions to tools like <b>Stable Diffusion</b>, <b>ChatGPT</b>, and <b>DALL-E</b> were linked with major increases in both revenue and collaboration. For instance, Germany’s Education sector saw an impact score of <b>54.27</b> when switching from Bard to Stable Diffusion — proving the strategic value of choosing the right AI tool for maximum business impact.
</div>


<hr style="border: none; border-top: 2px dashed #ccc; margin: 30px 0;" />


<!-- Title + Objective -->
<div style="font-family:'Segoe UI', sans-serif; margin: 20px 0;">

  <!-- Title Banner -->
  <div style="background: linear-gradient(to right, #f9a825, #fff176); padding: 12px 20px; border-radius: 8px;">
    <h2 style="color:#212121; margin:0;">⚖️ AI Content Volume vs Regulatory Readiness</h2>
  </div>

  <!-- Objective Block -->
  <div style="background-color:#fffde7; padding: 14px 20px; border-left: 5px solid #fbc02d; border-radius: 0 0 8px 8px;">
    <p style="margin:0;"><b>🎯 Objective:</b> To identify industry–country combinations that are generating high volumes of AI-generated content but operate in environments with weak or no regulatory oversight. These combinations may pose risks related to misinformation, ethical concerns, or ungoverned content generation.</p>
  </div>

</div>


In [23]:
query = """
WITH ContentVolumeStats AS (
  SELECT 
    Country,
    Industry,
    `Regulation Status`,
    ROUND(AVG(`AI-Generated Content Volume (TBs per year)`), 2) AS Avg_Content_TB,
    ROUND(AVG(`Consumer Trust in AI (%%)`), 2) AS Avg_Trust,
    COUNT(*) AS Entry_Count
  FROM ai
  GROUP BY Country, Industry, `Regulation Status`
),
RegulationTiered AS (
  SELECT *,
    CASE 
      WHEN UPPER(`Regulation Status`) = 'LENIENT' THEN 'No Regulation'
      WHEN UPPER(`Regulation Status`) = 'MODERATE' THEN 'Developing'
      WHEN UPPER(`Regulation Status`) = 'STRICT' THEN 'Strong'
      ELSE 'Unknown'
    END AS Reg_Tier
  FROM ContentVolumeStats
),
RiskyCombos AS (
  SELECT *,
         RANK() OVER (PARTITION BY Reg_Tier ORDER BY Avg_Content_TB DESC) AS Content_Rank_Within_Reg
  FROM RegulationTiered
  WHERE Reg_Tier IN ('No Regulation', 'Developing') AND Avg_Content_TB > 60
)
SELECT 
  Country,
  Industry,
  Reg_Tier,
  Avg_Content_TB,
  Avg_Trust,
  Entry_Count,
  Content_Rank_Within_Reg
FROM RiskyCombos
ORDER BY Reg_Tier, Content_Rank_Within_Reg
LIMIT 10;
"""
pd.read_sql(query, engine)


Country       Industry    Reg_Tier  Avg_Content_TB  Avg_Trust  \
0          USA     Automotive  Developing           96.13      54.47   
1        Japan          Legal  Developing           94.06      88.69   
2       France          Legal  Developing           93.76      51.84   
3        India          Legal  Developing           92.91      73.82   
4  South Korea          Legal  Developing           92.02      60.67   
5       Canada      Education  Developing           87.77      60.55   
6       Canada      Marketing  Developing           84.82      64.49   
7    Australia         Gaming  Developing           75.93      65.57   
8      Germany          Media  Developing           75.92      87.28   
9      Germany  Manufacturing  Developing           67.82      68.96   

   Entry_Count  Content_Rank_Within_Reg  
0            1                        1  
1            1                        2  
2            1                        3  
3            1                        4  
4            1                        5  
5            1                        6  
6            1                        7  
7            1                        8  
8            1                        9  
9            2                       10

<!-- Insight Block -->
<div style="font-family:'Segoe UI', sans-serif; background-color:#fffde7; padding: 14px 20px; border-left: 5px solid #f9a825; border-radius: 8px; margin-bottom: 30px;">
  <b>💡 Insight:</b> Countries like <b>USA (Automotive)</b> and <b>Japan (Legal)</b> are producing massive volumes of AI-generated content—<b>over 94 TBs per year</b>—while operating in regulatory environments classified as only “Developing.” This creates a high-risk scenario where content scale outpaces oversight, raising concerns around misinformation, ethics, and governance. Strikingly, these sectors also report high <b>consumer trust</b>, signaling a dangerous mismatch between public confidence and actual regulatory readiness.
</div>


<hr style="border: none; border-top: 2px dashed #ccc; margin: 30px 0;" />


<!-- Title + Objective -->
<div style="font-family:'Segoe UI', sans-serif; margin: 20px 0;">

  <!-- Title Banner -->
  <div style="background: linear-gradient(to right, #fbc02d, #fff59d); padding: 12px 20px; border-radius: 8px;">
    <h2 style="color:#212121; margin:0;">🎯 Trust & Market Share Growth Buckets</h2>
  </div>

  <!-- Objective Block -->
  <div style="background-color:#fffde7; padding: 14px 20px; border-left: 5px solid #fdd835; border-radius: 0 0 8px 8px;">
    <p style="margin:0;"><b>🎯 Objective:</b> Identify country–industry pairs showing significant changes in <b>Consumer Trust in AI (%)</b> and <b>Market Share of AI Companies (%)</b> from one year to the next. To ensure clarity and interpretability, percentage growth is categorized into growth tiers, helping avoid misinterpretations due to small base values.</p>
  </div>

</div>


In [24]:
query = """
WITH YearlyStats AS (
  SELECT
    Country,
    Industry,
    Year,
    ROUND(AVG(`Consumer Trust in AI (%%)`), 2) AS Avg_Trust,
    ROUND(AVG(`Market Share of AI Companies (%%)`), 2) AS Avg_Market_Share
  FROM ai
  GROUP BY Country, Industry, Year
),
WithPrevYear AS (
  SELECT
    curr.Country,
    curr.Industry,
    curr.Year,
    curr.Avg_Trust,
    prev.Avg_Trust AS Prev_Trust,
    curr.Avg_Market_Share,
    prev.Avg_Market_Share AS Prev_Market_Share
  FROM YearlyStats curr
  LEFT JOIN YearlyStats prev
    ON curr.Country = prev.Country
    AND curr.Industry = prev.Industry
    AND curr.Year = prev.Year + 1
),
Final AS (
  SELECT *,
    ROUND((Avg_Trust - Prev_Trust), 2) AS Trust_Abs_Growth,
    ROUND((Avg_Market_Share - Prev_Market_Share), 2) AS Market_Abs_Growth,
    ROUND(
      CASE 
        WHEN Prev_Trust > 0 THEN ((Avg_Trust - Prev_Trust) * 100.0) / Prev_Trust
        ELSE NULL 
      END, 2
    ) AS Trust_Perc_Growth,
    ROUND(
      CASE 
        WHEN Prev_Market_Share > 0 THEN ((Avg_Market_Share - Prev_Market_Share) * 100.0) / Prev_Market_Share
        ELSE NULL 
      END, 2
    ) AS Market_Perc_Growth
  FROM WithPrevYear
),
Bucketed AS (
  SELECT *,
    CASE 
      WHEN Trust_Perc_Growth >= 300 THEN 'Extreme Growth'
      WHEN Trust_Perc_Growth >= 100 THEN 'High Growth'
      WHEN Trust_Perc_Growth >= 50 THEN 'Moderate Growth'
      WHEN Trust_Perc_Growth >= 10 THEN 'Mild Growth'
      WHEN Trust_Perc_Growth IS NULL THEN 'N/A'
      ELSE 'Stable or Decline'
    END AS Trust_Growth_Tier,
    
    CASE 
      WHEN Market_Perc_Growth >= 300 THEN 'Extreme Growth'
      WHEN Market_Perc_Growth >= 100 THEN 'High Growth'
      WHEN Market_Perc_Growth >= 50 THEN 'Moderate Growth'
      WHEN Market_Perc_Growth >= 10 THEN 'Mild Growth'
      WHEN Market_Perc_Growth IS NULL THEN 'N/A'
      ELSE 'Stable or Decline'
    END AS Market_Growth_Tier
  FROM Final
)
SELECT 
  Country,
  Industry,
  Year,
  Avg_Trust,
  Prev_Trust,
  Trust_Abs_Growth,
  Trust_Perc_Growth,
  Trust_Growth_Tier,
  Avg_Market_Share,
  Prev_Market_Share,
  Market_Abs_Growth,
  Market_Perc_Growth,
  Market_Growth_Tier
FROM Bucketed
WHERE Trust_Perc_Growth IS NOT NULL OR Market_Perc_Growth IS NOT NULL
ORDER BY Trust_Perc_Growth DESC
LIMIT 15;
"""
pd.read_sql(query, engine)


Country       Industry  Year  Avg_Trust  Prev_Trust  Trust_Abs_Growth  \
0         Japan  Manufacturing  2024      77.96       34.52             43.44   
1         India         Gaming  2024      81.95       39.63             42.32   
2         China        Finance  2023      84.37       42.94             41.43   
3            UK         Gaming  2025      54.77       31.46             23.31   
4           USA         Retail  2021      68.14       44.56             23.58   
5     Australia     Automotive  2024      71.22       47.42             23.80   
6        France          Media  2022      78.71       55.15             23.56   
7       Germany  Manufacturing  2021      80.65       57.90             22.75   
8   South Korea          Legal  2025      79.04       60.67             18.37   
9        France          Media  2021      55.15       42.96             12.19   
10       France          Legal  2021      57.75       45.13             12.62   
11       France          Media  2024      72.79       58.85             13.94   
12           UK         Gaming  2022      66.38       54.27             12.11   
13    Australia         Gaming  2022      78.51       65.57             12.94   
14      Germany     Healthcare  2024      83.17       70.19             12.98   

    Trust_Perc_Growth Trust_Growth_Tier  Avg_Market_Share  Prev_Market_Share  \
0              125.84       High Growth              5.46              39.19   
1              106.79       High Growth              2.49              22.79   
2               96.48   Moderate Growth             14.69              36.32   
3               74.09   Moderate Growth              4.40              41.11   
4               52.92   Moderate Growth              8.09               3.46   
5               50.19   Moderate Growth             49.04              43.11   
6               42.72       Mild Growth             27.60              46.58   
7               39.29       Mild Growth              5.50              11.93   
8               30.28       Mild Growth              4.09              37.77   
9               28.38       Mild Growth             46.58              35.18   
10              27.96       Mild Growth             15.92               6.24   
11              23.69       Mild Growth             13.50              35.86   
12              22.31       Mild Growth             22.40              48.93   
13              19.73       Mild Growth             17.66               6.82   
14              18.49       Mild Growth             45.60              30.79   

    Market_Abs_Growth  Market_Perc_Growth Market_Growth_Tier  
0              -33.73              -86.07  Stable or Decline  
1              -20.30              -89.07  Stable or Decline  
2              -21.63              -59.55  Stable or Decline  
3              -36.71              -89.30  Stable or Decline  
4                4.63              133.82        High Growth  
5                5.93               13.76        Mild Growth  
6              -18.98              -40.75  Stable or Decline  
7               -6.43              -53.90  Stable or Decline  
8              -33.68              -89.17  Stable or Decline  
9               11.40               32.40        Mild Growth  
10               9.68              155.13        High Growth  
11             -22.36              -62.35  Stable or Decline  
12             -26.53              -54.22  Stable or Decline  
13              10.84              158.94        High Growth  
14              14.81               48.10        Mild Growth

<!-- Insight Block -->
<div style="font-family:'Segoe UI', sans-serif; background-color:#fff8e1; padding: 14px 20px; border-left: 5px solid #f9a825; border-radius: 8px; margin-bottom: 30px;">
  <b>💡 Insight:</b> Japan's Manufacturing industry experienced a <b>125.8%</b> surge in Consumer Trust but simultaneously saw a steep drop in AI Market Share, revealing a disconnect between public sentiment and actual industry influence. Meanwhile, countries like <b>France</b> (Legal, Media) and <b>Australia</b> (Gaming) achieved high growth in both trust and market presence, indicating successful ecosystem alignment. However, most top trust gainers faced a decline in market share, highlighting that trust alone doesn't guarantee economic dominance without broader AI company penetration.
</div>


<hr style="border: none; border-top: 2px dashed #ccc; margin: 30px 0;" />


<!-- Title + Objective -->
<div style="font-family:'Segoe UI', sans-serif; margin: 20px 0;">

  <!-- Title Banner -->
  <div style="background: linear-gradient(to right, #009688, #80cbc4); padding: 12px 20px; border-radius: 8px;">
    <h2 style="color:#ffffff; margin:0;">🎯 AI Consistency Champions: Country–Industry Pairs with Multi-Year Performance Above Global Standards</h2>
  </div>

  <!-- Objective Block -->
  <div style="background-color:#e0f2f1; padding: 14px 20px; border-left: 5px solid #00796b; border-radius: 0 0 8px 8px;">
    <p style="margin:0;"><b>🎯 Objective:</b> Identify which country–industry pairs have consistently outperformed <b>global averages</b> across key AI KPIs—<b>Adoption Rate</b>, <b>Revenue Impact</b>, and <b>Consumer Trust</b>—for multiple years. This helps surface reliable leaders that demonstrate sustained AI excellence, not just one-time success.</p>
  </div>

</div>


In [25]:
query = """
WITH GlobalAverages AS (
  SELECT 
    Year,
    ROUND(AVG(`AI Adoption Rate (%%)`), 2) AS Global_Adoption,
    ROUND(AVG(`Revenue Increase Due to AI (%%)`), 2) AS Global_Revenue,
    ROUND(AVG(`Consumer Trust in AI (%%)`), 2) AS Global_Trust
  FROM ai
  GROUP BY Year
),

Comparison AS (
  SELECT 
    d.Country,
    d.Industry,
    d.Year,
    CASE WHEN d.`AI Adoption Rate (%%)` > g.Global_Adoption THEN 1 ELSE 0 END AS Above_Adoption,
    CASE WHEN d.`Revenue Increase Due to AI (%%)` > g.Global_Revenue THEN 1 ELSE 0 END AS Above_Revenue,
    CASE WHEN d.`Consumer Trust in AI (%%)` > g.Global_Trust THEN 1 ELSE 0 END AS Above_Trust
  FROM ai d
  JOIN GlobalAverages g ON d.Year = g.Year
),

Aggregated AS (
  SELECT 
    Country,
    Industry,
    COUNT(DISTINCT Year) AS Total_Years,
    SUM(Above_Adoption) AS Years_Above_Adoption,
    SUM(Above_Revenue) AS Years_Above_Revenue,
    SUM(Above_Trust) AS Years_Above_Trust,
    (SUM(Above_Adoption) + SUM(Above_Revenue) + SUM(Above_Trust)) AS Consistency_Score
  FROM Comparison
  GROUP BY Country, Industry
),

Filtered AS (
  SELECT *
  FROM Aggregated
  WHERE (Years_Above_Adoption >= 3 AND Years_Above_Revenue >= 3)
     OR (Years_Above_Adoption >= 3 AND Years_Above_Trust >= 3)
     OR (Years_Above_Revenue >= 3 AND Years_Above_Trust >= 3)
)

SELECT *
FROM Filtered
ORDER BY Consistency_Score DESC
LIMIT 15;
"""
pd.read_sql(query, engine)


Country    Industry  Total_Years  Years_Above_Adoption  \
0       France       Media            6                   3.0   
1        India      Retail            2                   3.0   
2    Australia      Gaming            3                   3.0   
3      Germany       Media            3                   1.0   
4  South Korea       Legal            3                   2.0   
5           UK  Automotive            4                   3.0   
6           UK      Gaming            5                   4.0   
7       Canada      Gaming            3                   1.0   

   Years_Above_Revenue  Years_Above_Trust  Consistency_Score  
0                  4.0                2.0                9.0  
1                  2.0                4.0                9.0  
2                  2.0                3.0                8.0  
3                  4.0                3.0                8.0  
4                  3.0                3.0                8.0  
5                  3.0                2.0                8.0  
6                  3.0                1.0                8.0  
7                  3.0                3.0                7.0

<!-- Insight Block -->
<div style="font-family:'Segoe UI', sans-serif; background-color:#e0f7fa; padding: 14px 20px; border-left: 5px solid #00897b; border-radius: 8px; margin-bottom: 30px;">
  <b>💡 Insight:</b> <b>France–Media</b> and <b>India–Retail</b> emerge as the most consistent overachievers, each scoring a <b>Consistency Score of 9</b>, having surpassed global benchmarks in at least two out of three KPIs (AI adoption, revenue impact, trust) across multiple years. France–Media, with 6 years of data, notably excels in revenue (4 years above global avg) and adoption. This illustrates deep-rooted maturity in AI integration and sustained success. Other strong performers include <b>Australia–Gaming</b> and <b>UK–Automotive</b>, showing balanced, multi-dimensional AI impact over time.
</div>


<hr style="border: none; border-top: 2px dashed #ccc; margin: 30px 0;" />


<div style="border: 2px solid #a855f7; border-radius: 15px; padding: 20px; background: #faf5ff; font-family: 'Segoe UI', sans-serif; box-shadow: 0 4px 8px rgba(0,0,0,0.1);">

  <h2 style="color: #7e22ce;">🚀 Project Summary: Global AI Pulse (2020–2025)</h2>
  <p>
    This project was a high-speed ride through 10 countries and 10 industries, revealing how AI
    has evolved from 2020 to 2025. We used powerful SQL insights to uncover <strong>early adopters</strong>,
    <strong>economic impact</strong>, <strong>tool transitions</strong>, <strong>AI content booms</strong>,
    and even <strong>job loss paradoxes</strong>.
  </p>

  <h3 style="color: #a21caf;">🔥 Key Highlights:</h3>
  <ul>
    <li>🇮🇳 <strong>India’s Retail</strong> & 🇨🇳 <strong>China’s Manufacturing</strong> led the 2020 AI race.</li>
    <li>🇩🇪 <strong>Germany</strong> had the best revenue-per-AI-unit in 2024.</li>
    <li>🤖 Tool upgrades (e.g., to <code>ChatGPT</code> or <code>DALL·E</code>) triggered massive boosts.</li>
    <li>⚖️ Industries with <strong>high AI</strong> but <strong>declining job loss</strong>? That’s an AI paradox!</li>
    <li>🚨 We identified <strong>AI hotspots</strong> with big content volumes but low regulation—risky combo!</li>
    <li>🏆 Crowned <strong>AI Consistency Champions</strong> who stayed above global KPIs for years.</li>
  </ul>

  <p style="margin-top: 20px; font-style: italic;">
    With SQL magic, we made raw data tell global stories. 📈🌐
  </p>

  <p style="color: #6b7280; font-size: 14px; margin-top: 20px;">
    🔚 <strong>That’s a wrap!</strong> 
  </p>

</div>
